broker/broker-security-impl/src/main/java/org/codice/ddf/broker/security/impl/SubjectInjectorPlugin.java

```diff
@@ -122,6 +124,13 @@ public void clearCache() {
   Subject cacheAndReturnSubject(ServerSession session) throws SecurityServiceException {
     UPAuthenticationToken usernamePasswordToken =
         new UPAuthenticationToken(session.getUsername(), session.getPassword());
-    return securityManager.getSubject(usernamePasswordToken);
+    return AccessController.doPrivileged(
```

Comment:
✏️ I recommend this approach instead:
```java
  @VisibleForTesting
  Subject cacheAndReturnSubject(ServerSession session) throws SecurityServiceException {
    UPAuthenticationToken usernamePasswordToken =
        new UPAuthenticationToken(session.getUsername(), session.getPassword());
    try {
      return AccessController.doPrivileged(
          (PrivilegedExceptionAction<Subject>)
              () -> securityManager.getSubject(usernamePasswordToken));
    } catch (PrivilegedActionException e) {

      if (e.getCause() instanceof SecurityServiceException) {
        throw (SecurityServiceException) e.getCause();
      }
      throw new SecurityServiceException(e.getCause());
    }
  }
```

Reply:
is the `instanceof` check to prevent needing to create a new `SecurityServiceException`? why not just do 
`catch (PrivilegedExceptionAction e) {
   throw new SecurityServiceException(e.getCause());
}`
(also what's the benefit of doing `e.getCause()` vs `e`?)

Reply:
If it is already a `SecurityServiceException`, you don't need to create a new one and wrap it around another `SecurityServiceException`. If it's not (and, really, it _can't_ be anything else because any RT exceptions would bubble out but this is just the standard form for this), then you want to wrap whatever the underlying cause was in a new `SSException`.

You want the cause because the exception itself is a `PrivilegedExceptionAction` that wraps whatever the cause was. No need to carry that envelope around any longer.

Reply:
awesome, thanks for the explanation. fixed.